packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesAndAutocompleteMixin.ts

```diff
@@ -51,19 +58,19 @@ export class ApplicationCommandOptionWithChoicesAndAutocompleteMixin<T extends s
 	 *
 	 * @param choices The choices to add
 	 */
-	public addChoices(choices: [name: string, value: T][]): Omit<this, 'setAutocomplete'> {
+	public addChoices(...choices: ChoiceData<T>[]): Omit<this, 'setAutocomplete'> {
 		if (this.autocomplete) {
 			throw new RangeError('Autocomplete and choices are mutually exclusive to each other.');
 		}
 
 		choicesPredicate.parse(choices);
 
-		for (const [label, value] of choices) this.addChoice(label, value);
+		for (const { name, value } of choices) this.addChoice({ name, value });
 		return this;
 	}
 
-	public setChoices<Input extends [name: string, value: T][]>(
-		choices: Input,
+	public setChoices<Input extends ChoiceData<T>[]>(
+		...choices: Input
```

Comment:
I don't think you need this Input type generic

Reply:
it's a weird decision to add imo
but it has it's purpose, apparently previously if you had passed `setChoices([])` then it would still let you call `.setAutocomplete`, 
now of course if you pass nothing to it `setChoices()`, then it works the same, lets you call setAutocomplete
i am unaware of why this is added or why someone would call setChoices with an empty array but i didn't bother changing it